From: Shiraz Saleem <shirazsaleem@microsoft.com>
Date: Tue, 3 Sep 2024 22:05:59 -0500
Subject: providers/mana: Fix return values on unsupported parent domain flags

Return EOPNOTSUPP when unsupported flags are used to create parent
domain

Fixes: 443f196deee0 ("mana: Microsoft Azure Network Adapter (MANA) RDMA provider")
Signed-off-by: Shiraz Saleem <shirazsaleem@microsoft.com>
Origin: upstream, https://github.com/linux-rdma/rdma-core/pull/1556
---
 providers/mana/mana.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/providers/mana/mana.c b/providers/mana/mana.c
index efd5379..22e5ff9 100644
--- a/providers/mana/mana.c
+++ b/providers/mana/mana.c
@@ -113,7 +113,7 @@ mana_alloc_parent_domain(struct ibv_context *context,
 		verbs_err(
 			verbs_get_ctx(context),
 			"This driver supports IBV_PARENT_DOMAIN_INIT_ATTR_PD_CONTEXT only\n");
-		errno = EINVAL;
+		errno = EOPNOTSUPP;
 		return NULL;
 	}
 
